webofneeds/won-owner-webapp/src/main/resources/spring/owner-security.xml

```diff
@@ -90,13 +92,37 @@
 		<csrf disabled="true" />
 	</http>
 
+    <!-- because we use a customization of the RememberMeServices, 
+    we cannot use the <remember-me namespace config and have to define the required beans manually -->
+    
+    <beans:bean id="rememberMeFilter" class="org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter">
+        <beans:constructor-arg name="rememberMeServices" ref="rememberMeServices"/>
+        <beans:constructor-arg name="authenticationManager" ref="authenticationManager" />
+    </beans:bean>
 
+    <beans:bean id="rememberMeServices" class="won.owner.service.impl.KeystoreEnabledPersistentRememberMeServices">
+        <beans:constructor-arg name="userDetailsService" ref="wonUserDetailService"/>
+        <beans:constructor-arg name="key" value="${owner.webapp.rememberme.key}" />
+        <beans:constructor-arg name="tokenRepository" ref="tokenRepository"/>
+        <beans:property name="tokenValiditySeconds" value="5184000" />
```

Comment:
what kind of a token is this? so anonoumous accounts are valid 60 days, is this right?

Reply:
That's the remember-me token. All remember-me tokens for a user are deleted when the user clicks logout, and they lose validity after the set period.

However, they are not automatically purged from the database. We need to have a batch job do that on a regular basis. In our case, this is important, as this also clears the encrypted keystore passwords (which can be decrypted with the user's cookies).